src/Diagnostics.RuntimeHost/Controllers/Site/SitesController.cs

```diff
@@ -116,15 +182,65 @@ public async Task<IActionResult> GetInsights(string subscriptionId, string resou
             return await base.GetInsights(app, pesId, supportTopicId, startTime, endTime, timeGrain);
         }
 
+        /// <summary>
+        /// Publish detector package.
+        /// </summary>
+        /// <param name="pkg">The package.</param>
+        /// <returns>Task for publishing package.</returns>
         [HttpPost(UriElements.Publish)]
-        public async Task<IActionResult> PublishDetector(string subscriptionId, string resourceGroupName, string siteName, [FromBody] DetectorPackage pkg)
+        public async Task<IActionResult> PublishDetectorPackage([FromBody] DetectorPackage pkg)
         {
-            return await base.PublishDetector(pkg);
+            return await base.PublishPackage(pkg);
         }
 
-        private bool IsPostBodyMissing(DiagnosticSiteData postBody)
+        /// <summary>
+        /// Publish gist package.
+        /// </summary>
+        /// <param name="pkg">The package.</param>
+        /// <returns>Task for publishing package.</returns>
+        [HttpPost(UriElements.PublishGist)]
+        public async Task<IActionResult> PublishGistPackage([FromBody] GistPackage pkg)
```

Comment:
I wonder if it would make sense to create a separate controller, something like GistController. As far as I can understand, Gist do not rely on subscription, resourceGroup, sitename, or hosting environment input. What do you think @ShekharGupta1988 ?

Reply:
Yup. Its a good idea to create a GistController.

Ideally, we want a separation atleast at team level for Gists so that LogicApp is not messing up thr Web App gists or AKS gits.

So we can create a GistController with NameSpace as input ("Microsoft.Web" perhaps)

Reply:
I would like to keep the gist action here. We can use the resource type to filter gists.